### PR TITLE
Reapply "[libc][windows] start time API implementation (#117775)"

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
Reland #117775 with GPU fixes. Local build passed.

### CODE REVIEW

File: libc/src/__support/time/gpu/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  SRCS
+    time_utils.cpp
+  HDRS
+    time_utils.h
+  DEPENDS
+    libc.hdr.types.clock_t
+    libc.hdr.time_macros
+)
+
+add_entrypoint_object(
```
Comment (jhuber6) [inline libc/src/__support/time/gpu/CMakeLists.txt:12]: I really don't think it makes sense to have an entry point implementation live in `__support` Can we just make this an object library?

Comment (SchrodingerZhu) [reply libc/src/__support/time/gpu/CMakeLists.txt:12]: You are right. I `ctrl+c/ctrl+v`ed it from original place. Changing it now

Comment (SchrodingerZhu) [reply libc/src/__support/time/gpu/CMakeLists.txt]: fixed in https://github.com/llvm/llvm-project/pull/118886/commits/ea35f2a0cdc3b02d3299e49204d7b7fdd58a5456

**Review by jhuber6 — APPROVED** [review]

**Review by michaelrj-google — APPROVED** [review]

**Comment by SchrodingerZhu** [comment]
Does not seem to be my fault this time.

**Pull request merged**